src/sparse/node.rs

```diff
@@ -461,11 +481,11 @@ where
     StorageType::Error: fmt::Debug,
 {
     fn left_child(&self) -> Self {
-        StorageNode::left_child(self).unwrap()
+        StorageNode::left_child(self).unwrap().unwrap()
```

Comment:
Hmm, how do we know that it is safe? It seems I always can write the code like `node.left_child().left_child().left_child()` and get a panic

Reply:
Also maybe it is better to move logic from `StorageNode::left_child` and `StorageNode::right_child` to here(Into the `crate::common::ParentNode` implementation). It will remove duplication and will force the caller of to handle an error)

Reply:
> Hmm, how do we know that it is safe? It seems I always can write the code like node.left_child().left_child().left_child() and get a panic

You're right that this isn't exactly safe - if someone were to manually use the `StorageNode` and call a child getter, it may panic. It may also panic if the user manually changes the data in their database or corrupts their data in some way. It would be superior if `ParentNode` functions returned a `Result<Option<..>>` instead of requiring `unwrap`s. 

The challenge right now is that this would require some major changes to the way we use the `ParentNode` implementations, such as the [PathIter](https://github.com/FuelLabs/fuel-merkle/blob/master/src/common/path_iterator.rs#L159). I think this is definitely worth looking to, but I would like to do this work in a separate PR to keep the changes more focused.

> Also maybe it is better to move logic from StorageNode::left_child and StorageNode::right_child to here(Into the crate::common::ParentNode implementation). It will remove duplication and will force the caller of to handle an error)

Yes, that's fine too. I can refactor that when I work on updating the `ParentNode` return type.

Reply:
I created https://github.com/FuelLabs/fuel-merkle/issues/132 to track this issue.